test/yaml-rest-runner/src/main/java/org/elasticsearch/test/rest/yaml/section/MatchAssertion.java

```diff
@@ -97,6 +101,10 @@ protected void doAssert(Object actualValue, Object expectedValue) {
             assertThat(actualValue, instanceOf(List.class));
             assertMap((List<?>) actualValue, matchesList((List<?>) expectedValue));
         }
+        if (expectedValue instanceof String expectedValueAsString && actualValue instanceof byte[] actualValueAsBytes) {
```

Comment:
I had to add this here in order to be able to compare a byte array (like `_ignored_source`) using Base64 encoding. I don't see any other way to compare array of bytes.

Reply:
You can use this format in yaml and remove this change (it's pretty obscure feature but i tested it and it works).
```
- match: { hits.hits.0._ignored_source: !!binary "BgAAAG9iamVjdHktLS0KbmFtZTogImZvbyIKdmFsdWU6IDEwCg==" }
```

Reply:
I think we should use the obscure format instead of modifying the yaml assert code.

Reply:
Nice...I didn't know that